addons/mail/static/src/models/messaging_initializer.js

```diff
@@ -279,9 +279,11 @@ registerModel({
                     currentUser: insert({ id: currentUserId }),
                 });
             }
-            this.messaging.update({
-                partnerRoot: insert(this.messaging.models['Partner'].convertData(partner_root)),
-            });
+            if (partner_root) {
+                this.messaging.update({
+                    partnerRoot: insert(this.messaging.models['Partner'].convertData(partner_root)),
+                });
+            }
```

Comment:
I don't like these (as a result of not doing init messaging), as the code is currently not very robust to missing information like this one.

Reply:
I just guarded everything that was on `currentPartner`, as there is no `currentPartner` in frontend.
`currentUserGuest` doesn't work too, hence why guard.

All in all, it's too much effort to properly to everything at once.

The approach I take here is to have models available, even with few missing features (mostly due to special integration needed for frontend livechat specifically). When models are available and it's stable, then I can start moving a few LOCs at a time from legacy public livechat to models.

Reply:
`partnerRoot` is obtained with `init_messaging`, and I removed this feature for the time being in order to simply have the bare minimum of models to start refactoring JS livechat in small bits to models.

If I see things in current code of models that crash because it relies on `partnerRoot` being set, then I much prefer to fix it so that it does not need `partnerRoot` altogether.